src/passes/OptimizeCasts.cpp

```diff
@@ -104,8 +102,13 @@ struct BestCastFinder : public LinearExecutionWalker<BestCastFinder> {
   // This is tracked in each basic block, and cleared between them.
   std::unordered_map<Index, Expression*> mostCastedGets;
 
-  // For each most-downcasted local.get, a vector of other local.gets that could
-  // be replaced with gets of the downcasted value.
+  // Map local indices to the current downcasting of local.set to those indices.
+  //
+  // Also tracked in each basic block and cleared between them.
+  std::unordered_map<Index, Expression*> curCastedSets;
```

Comment:
It would be helpful to mention why there is only one current casted set for each local as opposed to having a  "most" casted set of many like we have for gets. As the reader, I initially expected to see a `mostCastedSets` map exactly mirroring `mostCastedGets`.

Also, we usually use `curr` rather than `cur` as an abbreviation of "current" in Binaryen.